netket/operator/_pauli_strings/jax.py

```diff
@@ -270,12 +269,31 @@ def _pauli_strings_mels_jax(local_states, z_data, x):
 
 
 @jax.jit
-def _pauli_strings_kernel_jax(local_states, x_flip_masks_all, z_data, x):
+def _pauli_strings_kernel_jax(local_states, x_flip_masks_all, z_data, x, cutoff=None):
+    mels = _pauli_strings_mels_jax(local_states, z_data, x)
+    if cutoff is not None:
+        nonzero_mels_mask = jnp.abs(mels) > cutoff
+        mels = jax.lax.select(nonzero_mels_mask, mels, jnp.zeros_like(mels))
+        # only flip if corresponding mel is nonzero
+        x_flip_masks_all = x_flip_masks_all & nonzero_mels_mask[..., None]
+    else:
+        nonzero_mels_mask = None
     # can re-use function from Ising
     x_prime = _ising_conn_states_jax(x[..., None, :], x_flip_masks_all, local_states)
-    mels = _pauli_strings_mels_jax(local_states, z_data, x)
-    # TODO do we want a cutoff?
-    return x_prime, mels
+    # TODO we could optionally move nonzeros to front here
+    return x_prime, mels, nonzero_mels_mask
+
+
+@jax.jit
+def _pauli_strings_n_conn_jax(local_states, x_flip_masks_all, z_data, x, cutoff=None):
+    _, _, nonzero_mels_mask = _pauli_strings_kernel_jax(
+        local_states, x_flip_masks_all, z_data, x, cutoff
+    )
+    if nonzero_mels_mask is not None:
+        return nonzero_mels_mask.sum(axis=-1, dtype=np.int32)
+    else:
+        max_conn_size = x_flip_masks_all.shape[0]
+        return jnp.full(x.shape[:-1], max_conn_size, dtype=np.int32)
```

Comment:
According to codecov the only tested branch in here is the first, where `nonezero_mels_mask is not None`. 
To be on the safe side, could we have a test exercising the other branch?

Reply:
I guess it's as simple as testing an operator with a `cutoff = None`